src/core/PointLight.ts

```diff
@@ -2,31 +2,36 @@ import {PointLight as ThreePointLight} from 'three'
 import LightBase from './LightBase'
 import {props} from './props'
 import {mapPropTo} from './props'
+import {prop} from '../html/WithUpdate'
 
 export default class PointLight extends LightBase {
     static defaultElementName = 'i-point-light'
 
-    static props = {
-        ...LightBase.props,
-        distance: mapPropTo({...props.number, default: 0}, (self: any) => self.three),
-        decay: mapPropTo({...props.number, default: 1}, (self: any) => self.three),
-        castShadow: mapPropTo({...props.boolean, default: true}, (self: any) => self.three),
-        shadowMapWidth: {...props.number, default: 512},
-        shadowMapHeight: {...props.number, default: 512},
-        shadowRadius: {...props.number, default: 3},
-        shadowBias: {...props.number, default: 0},
-        shadowCameraNear: {...props.number, default: 1},
-        shadowCameraFar: {...props.number, default: 2000},
-    }
-
+    @prop(mapPropTo({...props.number, default: 0}, (self: any) => self.three))
     distance!: number
+
+    @prop(mapPropTo({...props.number, default: 1}, (self: any) => self.three))
     decay!: number
+
+    @prop(mapPropTo({...props.boolean, default: true}, (self: any) => self.three))
     castShadow!: boolean
+
+    @prop({...props.number, default: 512})
     shadowMapWidth!: number
+
+    @prop({...props.number, default: 512})
     shadowMapHeight!: number
+
+    @prop({...props.number, default: 3})
     shadowRadius!: number
+
+    @prop({...props.number, default: 0})
     shadowBias!: number
+
+    @prop({...props.number, default: 1})
     shadowCameraNear!: number
+
+    @prop({...props.number, default: 2000})
     shadowCameraFar!: number
```

Comment:
Still some small duplication (f.e. `number` is seen twice in these props). I want to see if the following will work (and if needed update `@prop()` so that it works):

```ts
    @prop({...props.number})
    shadowCameraFar = 2000
```

instead of

```ts
    @prop({...props.number, default: 2000})
    shadowCameraFar!: number
```

Reply:
We can currently also use primitive constructors (f.e. `Number`) as shortcuts which are mapped to prop types (f.e. `props.number`). So we should be able to also write

```ts
    @prop(Number)
    shadowCameraFar = 2000
```

, which is the most concise.

Reply:
Also, I was debating decorator names. F.e.:

```ts
    @prop(Number)
    shadowCameraFar = 2000

    @type(Number)
    shadowCameraFar = 2000
```

or for a more complicated prop:

```ts
    @prop(mapPropTo({...props.number, default: 1}, (self: any) => self.three))
    decay!: number

    @type(mapPropTo({...props.number, default: 1}, (self: any) => self.three))
    decay!: number
```

Any other name ideas?

Reply:
If I recall, you were intending that the decorator removes the need for double declarations (which were used to set defaults? I haven't followed exactly why that was occurring). I do like the vertical compactness of the static props {} blocks that were there. I think it would be more pleasant if 
    @prop(Number)
    shadowCameraFar = 2000
can exist as a single-line statement of some sort, such as
    @prop(Number) shadowCameraFar = 2000
though that might not be achievable. It seems like in that case though, the type is being inferred, so I'm now not sure why the decorator was needed. At any rate, the type inference prevents a double type information being written. To make sure the compiler can always infer, perhaps there's a way to force the assignment of a default when using the decorator?

Reply:
@corruptedzulu Actually, `@prop(Number) shadowCameraFar = 2000` is totally valid syntax. :)

By the way, you just mentioned (pinged) GitHub user @prop in your comment. LOL

You can use back ticks to make code snippets, like this:

\`\`\`ts
if (true) foo()
\`\`\`

where `ts` means it will use TypeScript syntax highlight. You can specify `js`, `html`, etc.

Then when you post your comment, it will look like:

```ts
if (true) foo()
```

Reply:
I'll hadn't thought much about it. I'll update the statements to single-line for most of them. That'll be nicer. :)